device_independent_test/dimension.py

```diff
@@ -0,0 +1,60 @@
+from qiskit import QuantumCircuit
+
+def run_test(dispatcher, tolerance, shots):
+    """Runs dimensionality test for 2-qubits system. Alice prepares all possible
+    orthogonal states (no rotation), and Bob measures in corresponding canonical
+    states.
+
+    Args:
+        dispatcher (QuantumDispatcher)
+        tolerance (Number): passing tolerance
+        shots (int): number of shots to run
+
+    Returns:
+        Boolean: True if pass, False if fail
+    """
+    initial_states = [[0, 0], [0, 1], [1, 0], [1, 1]]
+    pre_ops = []
+
+    for state in initial_states:
+        pre_ops.append(prepare_bit_circuit(state))
+
+    post_ops = [[QuantumCircuit(2)], [measure_circuit()]]
+    
+    counts = dispatcher.batch_run_and_transmit(pre_ops, post_ops, shots)
+
+    success_prob = compute_success_probability(counts, shots)
+    passed = 1. - tolerance <= success_prob <= 1. + tolerance
+    return (passed, success_prob)
```

Comment:
This is probably fine as a tuple for now, but a dictionary would be more appropriate if we wanted there to be two dimension tests.